secure/storage/src/vault.rs

```diff
@@ -33,43 +37,65 @@ impl VaultStorage {
         token: String,
         namespace: Option<String>,
         certificate: Option<String>,
+        renew_ttl: Option<u32>,
     ) -> Self {
         Self {
             client: Client::new(host, token, certificate),
             namespace,
+            renew_ttl,
+            next_renewal: AtomicU64::new(0),
         }
     }
 
+    // Made into an accessor so we can get auto-renewal
+    fn client(&self) -> &Client {
```

Comment:
Given the new client() accessor, can we make the client field private?

Reply:
I think it should be private anyway and so should this function